### PR TITLE
allow multiple configs in yaml

### PR DESCRIPTION
**Pull request by Simontwice**
# Description
handle separating two configs in 1 yaml by ---
ALSO support yaml inheritance by setting a field in yaml: `parent: path/to/parent`

# Checklist
Answer each question with "yes", and provide an explanation if the answer isn't what is expected.

1. Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - Answer: not really but bear with me
2. Are all variables set with logical defaults that are backward compatible?
   - Answer: y
3. Can you provide a link to a Neptune experiment verifying the functionality of this PR?
   - Answer: no point
4. Have you successfully executed the linter and tests?
   - Answer: y
5. Are all functions concise and don't require splitting?
   - Answer: y
6. Is there documentation for arguments and complex logic?
   - Answer: y
7. Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - Answer: y
8. Are all functions placed in appropriate files?
   - Answer: y
9. Is the PR name meaningful and descriptive?
   - Answer: y
10. Is PR description provided, or unnecessary?
    - Answer: y
  
Other questions:

11. Have you not made changes to the requirements or anything related to the Singularity image? If you have, did you generate and upload a new image to the clusters?
    - Answer:
12. Is the change significant enough to notify all individuals working with this repository?
    - Answer:
13. Is there a need for a second reviewer?
    - Answer:

### CODE REVIEW

**Comment by sebastianjaszczur** [comment]
Please answer checklist questions. Especially I think that yaml validating the apporach could be run, with a link to Neptune.

File: lizrd/scripts/grid.py [hunk 1]
```diff
@@ -26,90 +27,88 @@
...
-
     try:
         path = sys.argv[1]
     except IndexError:
-        raise ValueError("config path must be specified as the first argument")
+        raise ValueError("No config path specified. Aborting...")
 
     if path.endswith(".yaml"):
-        grid_args = yaml.safe_load(open(sys.argv[1]))
+        configs = yaml.safe_load_all(open(sys.argv[1]))
```
Comment (sebastianjaszczur) [inline lizrd/scripts/grid.py:41]: what, does it work? how does it recognize end/beginning of yaml config? can you provide an example file with 2 or more configs?

**Review by sebastianjaszczur — COMMENTED** [review]
I would like to see an examle config. Also, I'd feel more comfortable if somebody else would run those changes and review them. But it LGTM from my side.

File: lizrd/scripts/grid.py (hunk 1 above)
Comment (Simontwice) [reply lizrd/scripts/grid.py]: see the diff now. Got tests! The engineering gods are smiling upon us

File: research/conditional/train/configs/test/test_baseline.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ params:
   tags:
     - "test"
   use_neptune: true
-  n_steps: 10
+  n_steps: 2
```
Comment (ludziej) [inline research/conditional/train/configs/test/test_baseline.yaml:18]: silent zmiana z 10 na 2, niezwiązana z PR, w efekcie nie stestuje się np. log heavy. Oprócz tego sam czas treningu jest mniejszą częscią czasu testu w porównaniu do uruchamiania environmentu wcześniej itp.

Comment (Simontwice) [reply research/conditional/train/configs/test/test_baseline.yaml]: zmienilem log heavy na 2
moj ADHD mozg nie moze tyle czekac

File: lizrd/support/misc.py [hunk 3]
```diff
@@ -24,3 +26,27 @@ def count_parameters(model, args, VOCAB_SIZE):
 def generate_random_string(length: int) -> str:
     letters = string.ascii_lowercase
     return "".join(random.choice(letters) for i in range(length))
+
+
+def load_with_inheritance(filepath, is_parent=False):
```
Comment (crewtool) [inline lizrd/support/misc.py]: Na pierwszy rzut oka nie jestem w stanie zrozumieć co ta funkcja robi. Mógłbyś napisać krótkiego docstriga?

File: research/conditional/tests/test_yaml_handling.py
Comment (crewtool) [inline research/conditional/tests/test_yaml_handling.py:1]: To wygląda jak test grida, a nie conditional

File: lizrd/scripts/grid.py [hunk 4]
```diff
@@ -25,92 +24,93 @@
...
+                "time",
+                "n_gpus",
+                "runner",
+                "cpus_per_gpu",
+                "nodelist",
+                "cuda_visible",
+            ],
+            [GRES, TIME, N_GPUS, RUNNER, CPUS_PER_GPU, NODELIST, CUDA_VISIBLE_DEVICES],
+        ):
+            PARAMS["temp_args"][name] = param
```
Comment (crewtool) [inline lizrd/scripts/grid.py:92]: Czemu nie możemy użyć tych parametrów bezpośrednio później, tylko tworzymy PARAMS["temp_args"]?

File: lizrd/support/misc.py (hunk 3 above)
Comment (Simontwice) [reply lizrd/support/misc.py]: ok, incoming

File: research/conditional/tests/test_yaml_handling.py
Comment (Simontwice) [reply research/conditional/tests/test_yaml_handling.py:1]: OK, a zaproponujesz gdzie je dać? bo ja nie wiem za bardzo, nie widzialem w `lizrd` testow

File: lizrd/scripts/grid.py (hunk 4 above)
Comment (Simontwice) [reply lizrd/scripts/grid.py]: powod taki, ze to jest wczytywane z configa, i dla kazdego configa moze byc inaczej. Zgadzam się, że to może być napisane lepiej. Sugestie appreciated.

File: research/conditional/tests/test_yaml_handling.py
Comment (Simontwice) [reply research/conditional/tests/test_yaml_handling.py:1]: done

File: lizrd/support/misc.py (hunk 3 above)
Comment (Simontwice) [reply lizrd/support/misc.py]: done

**Pull request closed without merging**